proxy/http/HttpSM.cc

```diff
@@ -5562,7 +5565,9 @@ HttpSM::do_setup_post_tunnel(HttpVC_t to_vc_type)
     //  header buffer into new buffer
     client_request_body_bytes = post_buffer->write(ua_buffer_reader, chunked ? ua_buffer_reader->read_avail() : post_bytes);
 
-    ua_buffer_reader->consume(client_request_body_bytes);
+    // Note: We can not consume the data sice we start retry.
+    // ua_buffer_reader will be destroy in HttpSessionClient::free
+    // ua_buffer_reader->consume(client_request_body_bytes);
```

Comment:
The ua_buffer_reader only keeps the first part of request payload. If the tunnel has read the second part of the request payload, the ATS will only replay the first part of request payload when retry to another server and the second part will be lost.